include/modem/lte_lc.h

```diff
@@ -1457,26 +1457,34 @@ int lte_lc_edrx_req(bool enable);
  *
  * For reference see 3GPP 24.301 Ch. 9.9.4.25.
  *
+ * @deprecated Use AT%RAI and socket options NRF_SO_RAI_* instead.
```

Comment:
`%` sign not visible in generated documentation.

And then thinking aloud.
I'm still thinking if we should have a function for RAI... We should check if there are similar examples. Maybe it's enough that the use of AT%RAI is mentioned in the NRF_SO_RAI_* descriptions properly.

Reply:
I'll check the percentage sign.

@trantanen We could have a function for `AT%RAI`, one benefit would be that there could be a brief documentation how the feature is used. However, logical name for the function would be `lte_lc_rai_req()`, which currently uses `AT%XRAI` and is being deprecated. The function would have to be named differently or then we could just change the functionality, but not sure if that's ok.

@lemrey @simensrostad @jtguggedal  Do you have any insight, could we just update `lte_lc_rai_req()` to use the new RAI functionality? I'm just worried that developers could miss the change and not notice that they need to take the socket options into use. `lte_lc_rai_param_set()` would still be deprecated.

Reply:
If we think there are customers using RAI functionality with the existing `lte_lc_rai_req()` function, I'd be a bit reluctant in reusing the same function for the `AT%RAI` command.

Reply:
@trantanen I have a proposal for this. I don't think there's need to change `AT%RAI` value at runtime, so the logical thing to do is to add a new Kconfig option for it. The name can be `CONFIG_LTE_RAI_REQ`, which is unused and in line with other similar options. Both `lte_lc_rai_req()` and `lte_lc_rai_param_set()` can be deprecated and still use `AT%XRAI`.

Reply:
Without checking all details, sounds like a plan.

Reply:
I agree about the Kconfig option. If needed to be toggled at runtime, it's a one API call anyway; we could have a Kconfig option that sets it automatically. One question is whether the `AT%RAI` setting preserved on `CFUN` changes or is it reset?

Reply:
@lemrey `AT%RAI` setting is stored into modem NVM. I implemented this so, that `AT%RAI` command is always sent at boot time. So, depending on `CONFIG_LTE_RAI_REQ`, RAI is always explicitly enabled or disabled at boot. Also PSM and eDRX work like this.